### DISCUSSION TITLE
Deriving Lineage from Spark Event Logs?

### DISCUSSION THREAD

**Discussion by butikiy** [OpenLineage/OpenLineage#1877] [Q&A]
Hi All,

I would like to know if it is possible to

- Derive Table and column level lineage from Spark Event logs (spark.eventLog.enabled=true) which are persisted (spark.eventLog.dir=). This means that a custom listener like open-lineage would not need to be registered in run time and this saves us from thinking about/answering questions like memory footprint of the listener, its impact on the actual spark application etc. I looked at the event logs and see that "SparkListenerSQLExecutionStart" event seems promising as it contains the Physical/Logical plan info. Please let me know if I am thinking in the right direction.

![Screenshot 2023-05-24 at 1 15 43 PM](https://github.com/OpenLineage/OpenLineage/assets/134486845/cf9c7adf-1ecd-46b7-91c5-60b21d99466e)

- If the above point is possible, Would I be able to use open-lineage in some fashion to gather the Lineage from persisted Event logs ? ( For example- in a batch job)

**Comment by mobuchowski** [comment]
Probably not - with being registered at runtime, we have access to way larger context and are able to provide way more accurate information than pure string representation of LogicalPlan would give us. Possibly it would be different for your use case - if you have to care about single source/sink type, etc - but in general case I believe it's close to impossible.
